module/core.cppm

```diff
@@ -0,0 +1,207 @@
+// Copyright (c) 2025 Antony Polukhin
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+// To compile manually use a command like the folowing:
+// clang++ -I ../include -std=c++20 --precompile -x c++-module core.cppm
+
+#define BOOST_CORE_BEGIN_MODULE_EXPORT export {
+#define BOOST_CORE_END_MODULE_EXPORT }
+
+module;
+
+#include <boost/config/header_deprecated.hpp>
+#include <boost/config.hpp>
+#include <boost/config/no_tr1/memory.hpp>
+#include <boost/config/pragma_message.hpp>
+#include <boost/config/workaround.hpp>
+
+#include <boost/assert.hpp>
+#include <boost/assert/source_location.hpp>
+
+#include <boost/current_function.hpp>
+#include <boost/static_assert.hpp>
+
+#include <boost/throw_exception.hpp>  // TODO: modularize
+
+#if (defined(__cpp_lib_uncaught_exceptions) && __cpp_lib_uncaught_exceptions >= 201411)
+#if defined(__APPLE__)
+#include <Availability.h>
+#endif
+#endif
+
+#ifndef BOOST_CORE_HAS_STD_MODULE
+#include <algorithm>
+#include <array>
+#include <cctype>
+#include <climits>
+#include <cmath>
+#include <cstdarg>
+#include <cstddef>
+#include <cstdio>
+#include <cstdlib>
+#include <cstring>
+#include <cxxabi.h>
+#include <exception>
+#include <functional>
+#include <initializer_list>
+#include <ios>
+#include <iosfwd>
+#include <iostream>
+#include <iterator>
+#include <limits>
+#include <memory>
+#include <new>
+#include <stdexcept>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string>
+#include <time.h>
+#include <typeinfo>
+#include <type_traits>
+#include <utility>
+#include <wchar.h>
+#endif
+
+export module Boost.Core;
+
+#ifdef BOOST_CORE_HAS_STD_MODULE
```

Comment:
This macro should be changed to some Boost.Config macro that is defined if the compiler understands modules and there is module std. The includes in the headers should be guarded by that macro and use `import std; ` if it is avalable